### ISSUE TITLE
[Todo] More getters (e.g. variants, reds?)

### ISSUE THREAD

**Issue by zaydek** [zaydek/duomo#227]
Right now we have getter functions that are useful with the Sass API. They make it easy to get `red-500` or `weight(bold)`. The library has a `getters` folder and inside of it are these functions. That way if someone uses Duomo with Sass, they can reference these functions.

It would be nice if we had getters for color tones, like `reds()`, `purples()`, etc. These would look at `src/configuration/colors.scss` and return only some of the colors. You could use something like your `strSlice` idea or the `str-starts-with` function in helpers to check if the colors starts with `green`, etc. and only return a map with those values.

So basically this is the API:

```scss
reds();
// ( red-100: <hsl color>, red-200: <hsl color>, ... );

greens();
// ( green-100: <hsl color>, green-200: <hsl color>, ... );
```

**Comment by zaydek** [comment]
Also missing variants possibly.

**Comment by zaydek** [comment]
Might be nice if we even have like `reds`, etc. The `-vars` may be overkill.

**Comment by zaydek** [comment]
`cursors`?

**Comment by zaydek** [comment]
Once you get that basic idea working, there are two more upgrades after that:

We want to be able to pass an alpha channel as a value, so like:

```scss
reds(0.5)
// ( red-100: <hsla color with 0.5 alpha>, red-200: <hsla color with 0.5 alpha>, ... );

greens(0.5)
// ( green-100: <hsla color with 0.5 alpha>, green-200: <hsla color with 0.5 alpha>, ... );
```

**Comment by zaydek** [comment]
Finally, it might be nice if there was a more general API. So instead of `reds()`, maybe we just have like `color-matches(red)` and then it only returns the red colors. This way we don’t need a getter per `reds`, `blues`, but just one function.

```scss
color-matches(red, 0.5);
// ( red-100: <hsla color with 0.5 alpha>, red-200: <hsla color with 0.5 alpha>, ... );

color-matches(geen, 0.5);
// ( green-100: <hsla color with 0.5 alpha>, green-200: <hsla color with 0.5 alpha>, ... );
```

So if you have a  `color-matches` function, then it’s very very easy to write `reds` because it just calls `color-matches` instead.
